protected.go

```diff
@@ -264,11 +241,14 @@ func (p *Protector) DialUDP(network string, laddr, raddr *net.UDPAddr) (*net.UDP
 // unnecessary work, but doesn't support arbitrary cancellation.
 func (p *Protector) dialContext(op ops.Op, ctx context.Context, network, addr string) (net.Conn, error) {
 	var socketType int
+	var proto int
 	switch network {
 	case "tcp", "tcp4", "tcp6":
 		socketType = syscall.SOCK_STREAM
+		proto = syscall.IPPROTO_TCP
 	case "udp", "udp4", "udp6":
 		socketType = syscall.SOCK_DGRAM
+		proto = syscall.IPPROTO_UDP
```

Comment:
Using the proto explicitly is not strictly needed, but I added it while debugging other stuff and it doesn't hurt anything to be precise.